android/intltest/java/com/facebook/hermes/test/HermesIntlAndroidTest.java

```diff
@@ -17,14 +17,38 @@
 import org.junit.Test;
 
 public class HermesIntlAndroidTest extends InstrumentationTestCase {
+
+  private void evalScriptFromAsset(JSRuntime rt, String filename) {
+    try {
+      AssetManager assets = getInstrumentation().getContext().getAssets();
+      InputStream is = assets.open(filename);
+      String script = new BufferedReader(new InputStreamReader(is))
+              .lines().collect(Collectors.joining("\n"));
+      rt.evaluateJavaScript(script);
+    } catch (IOException ex) {
+      ex.printStackTrace();
+      assert (false);
+    }
+  }
+
   @Test
   public void testIntlFromAsset() throws IOException {
-    AssetManager assets = getInstrumentation().getContext().getAssets();
-    InputStream is = assets.open("intl.js");
-    String script = new BufferedReader(new InputStreamReader(is))
-      .lines().collect(Collectors.joining("\n"));
     try (JSRuntime rt = JSRuntime.makeHermesRuntime()) {
-      rt.evaluateJavaScript(script);
+      evalScriptFromAsset(rt, "intl.js");
+    }
+  }
+
+  @Test
+  public void testIntlGetCanonicalNamesTestScriptFromAsset() throws IOException {
+    try (JSRuntime rt = JSRuntime.makeHermesRuntime()) {
+      evalScriptFromAsset(rt, "sta.js");
```

Comment:
I am concerned that this approach to testing will not scale.

Hermes currently has infrastructure to run against test262, but we expect developers to download the tests separately.  See https://github.com/facebook/hermes/tree/master/utils/testsuite.

In this case, it's a little different, since we need the tests to be around at build time, but there are more than 800 files in `test262/test/intl402`, and copying them all into the Hermes repo does not feel like the right answer.

A better solution would be to have the build system take a path to the testsuite as an option.  We already maintain a list of tests which we expect Hermes will not pass; we can maintain a second one for Intl.  (We can't use the same one, as all Intl tests need to be skipped on platforms without Intl.)

Reply:
Hi @mhorowitz ,

Sorry for a late response .. It's been quite a hectic last couple of weeks both at home and work :)

I'm back now and am making some progress fixing many of the corner cases following Firefox's implementation .. which in turn will make it easier for us to have a low overhead blacklist based tests ... I'll update this PR soon with those changes along with some performance fixups ..

Reply:
Hi @mhorowitz 
We do have a blacklist based test coverage in the latest iteration. We've added the test files from test262 as it is without any changes, and apply the blacklist. It would be better to download the tests on the fly, but i'd like to go ahead checking in those, for now.

In order to be able to formalize the test as this, I had to make the implementation much richer. Currently we have a formal statement/definition on what do we don't support, for instance.